### PR TITLE
Bump GE plugin to 3.15.1

### PR DESCRIPTION
**Pull request by jean-andre-gauthier**

### CODE REVIEW

**Comment by jean-andre-gauthier** [comment]
@bot-gradle test this

File: gradle/verification-metadata.xml [hunk 1]
```diff
@@ -641,6 +639,11 @@
             <pgp value="31bae2e51d95e0f8ad9b7bcc40a3c4432bd7308c"/>
          </artifact>
       </component>
+      <component group="com.gradle" name="gradle-enterprise-gradle-plugin" version="3.15.1">
+         <artifact name="gradle-enterprise-gradle-plugin-3.15.1.jar">
+            <pgp value="7b79add11f8a779fe90fd3d0893a028475557671"/>
+         </artifact>
+      </component>
```
Comment (wolfs) [inline gradle/verification-metadata.xml:646]: ❌ I am somewhat surprised by this. Is there a new signing key for the Gradle Enterprise plugin? If yes, it should be updated in the trusted key entry above.

Also, can we remove the `com.gradle` entries now from the trusted key `1bd97a6a154e7810ee0bc832e2f38302c8075e3d` above? If we can't, were do we have artifacts that use the old signing key?

**Review by wolfs — CHANGES_REQUESTED** [review]

File: gradle/verification-metadata.xml (hunk 1 above)
Comment (jean-andre-gauthier) [reply gradle/verification-metadata.xml]: Hi @wolfs, thanks very much for your review. Afaiu we're indeed signing the GE plugin with a new key. I've removed the old `com.gradle` entries, and moved the `gradle-enterprise-gradle-plugin` key to a trusted key. Please let me know if this looks fine to you?

File: gradle/verification-metadata.xml [hunk 2]
```diff
@@ -82,13 +82,12 @@
          <trusted-key id="0a123c1ed3f13a6a0140e166c71fb765cd9de313" group="org.apache.ant"/>
          <trusted-key id="0cb5871fb7bf3b351614bbf6ca85ffe638d4407a" group="it.unimi.dsi" name="fastutil"/>
          <trusted-key id="0e18eae07b7774eac5db3f2113bb90ce8eafbe37" group="com.microsoft.playwright"/>
+         <trusted-key id="7b79add11f8a779fe90fd3d0893a028475557671" group="com.gradle" name="gradle-enterprise-gradle-plugin"/>
```
Comment (wolfs) [inline gradle/verification-metadata.xml:85]: 🤔 I'd probably have replaced the section for the trusted key for `com.gradle` with the new key id, though I am also fine to do it like this. We only need to remember to give more trust to this key should we use more artifacts from the `com.gradle` group.

```
         <trusted-key id="0ef2cac6b3577539b7ab5bcc91aa2c186c4d7a86">
            <trusting group="com.gradle"/>
         </trusted-key>
```

**Review by wolfs — APPROVED** [review]

**Comment by wolfs** [comment]
@bot-gradle test and merge

**Pull request merged**